### ISSUE TITLE
Allow wildcards in asset list

### ISSUE THREAD

**Issue by theshane** [jhthorsen/mojolicious-plugin-assetpack#50]
One thing that I like about Grunt is the ability to point it at a dir and it will run the functions on all files in the dir. That would be great to have here as well. I did not see a way to do this so I am just getting the file list in my app and then pushing those files onto the array. 

Looking at the code, I can see why *.js may not work out of the box (unless I totally missed something) but it would be nice to have. 

This is how I am handling it right now:

my @js_files = (
      "https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.js",
      "https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.4/js/bootstrap.js",
      "https://cdnjs.cloudflare.com/ajax/libs/angular.js/1.3.15/angular.js",
      "https://cdnjs.cloudflare.com/ajax/libs/angular.js/1.3.15/angular-route.js",
      "https://cdnjs.cloudflare.com/ajax/libs/highcharts/4.1.5/highcharts.js",
    );

  my $js_dir = $self->config('path')."/public/js/src";

  my @ls_js_dir = `ls $js_dir`;

  foreach my $file(@ls_js_dir){
  	chomp($file);
  	push @js_files, 'js/src/'.$file;
  }

  $self->app->asset(
    "app.js" => @js_files
  );

**Comment by zoffixznet** [comment]
How would the order of files be defined when using such a feature, if implemented? The order is significant for both CSS and JS.

**Comment by theshane** [comment]
Good question,

I have had this problem and Grunt will allow you to list files as well as a directory. It will not double dip on files so you explicitly call the ones that need to go first in the list.

I did have to do this in my app today:

my @js_files = (
      "https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.js",
      "https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.4/js/bootstrap.js",
      "https://cdnjs.cloudflare.com/ajax/libs/angular.js/1.3.15/angular.js",
      "https://cdnjs.cloudflare.com/ajax/libs/angular.js/1.3.15/angular-route.js",
      "https://cdnjs.cloudflare.com/ajax/libs/highcharts/4.1.5/highcharts.js",
      "https://raw.githubusercontent.com/pablojim/highcharts-ng/master/dist/highcharts-ng.js",
      "js/src/init.js"
    );

  my $js_dir = $self->config('path')."/public/js/src";

  my @ls_js_dir = `ls $js_dir`;

  foreach my $file(@ls_js_dir){
    chomp($file);
  	push @js_files, 'js/src/'.$file if($file ne 'init.js');
  }


  $self->app->asset(
    "app.js" => @js_files
  );

At some point I will write something that checks if the file is in the array or not and does not push it.

**Comment by jhthorsen** [comment]
@zoffixznet: Not always. Some js files doesn't care about order. I got this code in one of my project:
```
sub _js_assets {
  my $self = shift;
  # order is important
  my @js   = qw(
    /js/materialize/hammer.min.js
    https://code.jquery.com/jquery-2.1.1.min.js
    https://muut.com/riotjs/dist/demo/js/riot.js
    /js/materialize/velocity.min.js
    /js/materialize/jquery.hammer.js
    /js/materialize/jquery.easing.1.3.js
    /js/materialize/jquery.hammer.js
    /js/materialize/jquery.timeago.min.js
    /js/materialize/buttons.js
    /js/materialize/collapsible.js
    /js/materialize/dropdown.js
    /js/materialize/forms.js
    /js/materialize/sideNav.js
    /js/materialize/tooltip.js
    /js/materialize/waves.js
    /js/router.js
    /js/utils/window.js
  );

  # not important
  for my $dir (qw( mixins tags )) {
    for my $file (@{$self->home->list_files("public/js/$dir")}) {
      push @js, "/js/$dir/$file" if $file =~ m!\.(js|tag)$!;
    }
  }

  return @js, '/js/main.js';
}
```

I do believe that such a feature should have a consistent order though, so instead of simply reading the directory, I would have a ```sort {$a cmp $b}``` in there to make it consistent. That way, you could also name your files 00-very-important.js ... 99-load-me-last.js.

I'm still not convinced that it's a core feature, but I sure are starting to think so.

**Comment by theshane** [comment]
I use Angular primarily and the order does not matter except it is good to have the module creation in the beginning. That being said, if order does matter then I would either name the files that would allow sorting to do it's job or create different folders for each type of file and put those folders in order.

js/setup
js/services
js/controllers
js/directives

**Comment by jhthorsen** [comment]
I'm kind of swamped after the sprites addition. I still haven't managed to document nor make a proper API, so for now this is the deal:

If anyone can make a PR with a documentation patch for the wildcard feature, I will implement the logic. If not, I'm afraid this have to wait.

**Comment by theshane** [comment]
I completely understand. I was going to try to get it done before posting this but I too am limited on time. If I get some free time I will give it a shot and send it over though. 

Thanks again for building this. I was sick of including node files(Grunt,Bower...) in my perl apps :).

**Comment by jhthorsen** [comment]
You're very welcome :)

By the way, the PR doesn't have to be complete. The most important part is the documentation.

**Comment by theshane** [comment]
I agree.

**Comment by jhthorsen** [comment]
Closed by 5b817db. Please re-open this issue if it doesn't work as expected.

This code is already available on CPAN, in version 0.56.
